components/studentPortal/freemium/FreemiumDialogComponent.tsx

```diff
@@ -0,0 +1,105 @@
+import { XIcon } from "@heroicons/react/outline";
+import { Close, Content, Overlay, Portal, Root } from "@radix-ui/react-dialog";
+import React, { useEffect, useState } from "react";
+import { useSelector } from "react-redux";
+import { getModalContent } from "../../../pages/api/studentPortal/freemium/getModalContent";
+import { themeSelector } from "../../../redux/themeSlice";
+import TrailAnimation from "../../ui/TrailAnimation";
+
+export enum ModalStage {
+  ONE = 1,
```

Comment:
this can just be ModalStage = {ONE, TWO, THREE...}, enums have indexes automatically... not sure what you're doing here

Reply:
cleaned this up.